### PR TITLE
[WIP] feat: Smallest functional test for chainPR

### PR DESCRIPTION
**Pull request by yoshwata**
## Context

This PR adds smallest functional tests to keep the quality of chainPR

## Objective

This PR adds a new test for chainPR. Details can be found in the feature file.

However, it doesn't work because it was intentionally disabled.

In order to run it, [testOrg](https://github.com/screwdriver-cd-test) needs a new repository and screwdriver.yaml. The repository name supposed to be `functional-chainPR`

After being merged, authorized members should add the repository and screwdriver.yaml and remove [`@ignore`](https://github.com/screwdriver-cd/screwdriver/compare/master...sonic-screwdriver-cd:add-chainpr-test?expand=1#diff-4e0f06952f23b6671797dca68cb8fb191d0b29063f14f140a5805cb82f4e87f6R144).
Also, you can add the repository first. In that case, `ignore` is removed in this PR

The minimum chainPR configuration is as follows, and it's also supposed to be yaml in the `functional-chainPR` repo.
```
annotations:
  screwdriver.cd/chainPR: true

shared:
    image: node:14
jobs:
    SIMPLE:
        steps:
            - echo: echo test
        requires: [~pr, ~commit]
    AFTER-SIMPLE:
        steps:
            - echo: echo test
        requires: [SIMPLE]
```

The workflow is illustrated below.
functional-chainPR-graph

I have confirmed that workflow tests pass in my local environment.

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by yoshwata** [comment]
I will fix lint

File: features/workflow.feature [hunk 1]
```diff
@@ -140,3 +140,15 @@ Feature: Workflow
         And the "SIMPLE" job is not triggered
         And the "MASTER-PR" job is not triggered
         And that "REGEX" build uses the same SHA as the "STAGING" build
+
+    @ignore @workflow-chainPR
+    Scenario: chainPR
+        Given an existing pipeline on "master" branch with the workflow jobs:
+            | job       | requires  |
+            | SIMPLE    | ~commit   |
```
Comment (wahapo) [inline features/workflow.feature:148]:
```suggestion
            | SIMPLE    | ~pr       |
```

Comment (yoshwata) [reply features/workflow.feature]: Thank you. Fixed with table misalignment.

**Review by wahapo — APPROVED** [review]

File: features/workflow.feature [hunk 2]
```diff
@@ -140,3 +140,15 @@ Feature: Workflow
         And the "SIMPLE" job is not triggered
         And the "MASTER-PR" job is not triggered
         And that "REGEX" build uses the same SHA as the "STAGING" build
+
+    @ignore @workflow-chainPR
+    Scenario: chainPR
+        Given an existing pipeline on "master" branch with the workflow jobs:
```
Comment (tkyi) [inline features/workflow.feature:146]: should we change this to `"main" branch`? I know GitHub defaults to `main` branch moving forward

I'm open to changing the repo default branch to "master" if you think that makes more sense

**Review by tkyi — COMMENTED** [review]
Created test repo with yaml config: https://github.com/screwdriver-cd-test/functional-chainPR
Also invited you as Owner for screwdriver-cd-test repo

Just need to figure out default branch name

**Comment by yoshwata** [comment]
@tkyi Since some APIs are based on master, this time I will proceed with the default as master. I've changed default branch as master about `functional-chainPR` repo.

https://github.com/screwdriver-cd/screwdriver/blob/250169277acf950238b94adfa7fb7dbebb1f1304/features/support/github.js#L94

https://github.com/screwdriver-cd/screwdriver/blob/250169277acf950238b94adfa7fb7dbebb1f1304/features/support/github.js#L151

Thank you for your perfect work. I've removed `@ignore`.

**Comment by tk3fftk** [comment]
related: https://github.com/screwdriver-cd/screwdriver/issues/1393

File: features/workflow.feature [hunk 3]
```diff
@@ -92,6 +92,18 @@ Feature: Workflow
         And the "SIMPLE" job is not triggered
         And that "REGEX" build uses the same SHA as the "STAGING" build
 
+    @workflow-chainPR
+    Scenario: chainPR
+        Given an existing pipeline on "main" branch with the workflow jobs:
```
Comment (tkyi) [inline features/workflow.feature:97]:
```suggestion
        Given an existing pipeline on "master" branch with the workflow jobs:
```

File: features/support/sdapi.js [hunk 4]
```diff
@@ -18,6 +18,31 @@ function promiseToWait(timeToWait) {
...
+ * @param  {Object}  config                     Configuration object
+ * @param  {String}  config.instance            Screwdriver instance to test against
+ * @param  {String}  config.pipelineId          Pipeline ID to find the build in
+ * @return {Promise}                            A promise that resolves to an array of builds that
+ *                                              fulfill the given criteria. If nothing is found, an
+ *                                              empty array is returned
+ */
+function findJobs(config) {
+    const { instance } = config;
+    const { pipelineId } = config;
```
Comment (tk3fftk) [inline features/support/sdapi.js:34]: small refactor
```suggestion
    const { instance, pipelineId } = config;
```

**Review by tk3fftk — APPROVED** [review]
:+1:

**Review by kumada626 — APPROVED** [review]

**Comment by yoshwata** [comment]
We have found that this test has a high probability of failure in our environment. I will investigate.

**Comment by yoshwata** [comment]
Closed for rethinking.

**Pull request closed without merging**